pkg/cdi/spec.go

```diff
@@ -262,3 +262,56 @@ func validateSpec(raw *cdi.Spec) error {
 	}
 	return nil
 }
+
+// GetSpecFileBase returns a Spec file basename.
+func GetSpecFileBase(vendor, class, tag string) string {
+	if tag != "" {
+		return vendor + "-" + class + "-" + tag
+	}
+	return vendor + "-" + class
+
+}
+
+// GenerateSpecName generates a vendor+class scoped Spec file name. The
+// name can be passed to WriteSpec() to write a Spec file to the file
+// system.
+//
+// vendor and class should match the vendor and class of the CDI Spec.
+// The file name is generated without a ".json" or ".yaml" extension.
+// The caller can append the desired extension to choose a particular
+// encoding. Otherwise WriteSpec() will use its default encoding.
+//
+// This function always returns the same name for the same vendor/class
+// combination. Therefore it cannot be used as such to generate multiple
+// Spec file names for a single vendor and class.
+func GenerateSpecName(vendor, class string) string {
+	return vendor + "-" + class
+}
+
+// GenerateTransientSpecName generates a vendor+class scoped transient
+// Spec file name. The name can be passed to WriteSpec() to write a Spec
+// file to the file system.
+
+// Transient Specs are those whose lifecycle is tied to that of some
+// external entity, for instance a container. vendor and class should
+// match the vendor and class of the CDI Spec. driver can be left empty
+// if only a single CDI user is generating transient Spec files for the
+// given vendor/class combination on the host. Otherwise driver should
+// uniquely identify the caller among those that use the same vendor
+// and class. transientID should uniquely identify the external entity
+// within the scope of the caller/driver.
+//
+// The file name is generated without a ".json" or ".yaml" extension.
+// The caller can append the desired extension to choose a particular
+// encoding. Otherwise WriteSpec() will use its default encoding.
+func GenerateTransientSpecName(vendor, class, driver, transientID string) string {
+	name := vendor + "-" + class
+
+	if driver != "" {
+		name += "+" + driver
+	}
+
+	name = "transient." + name + "_" + transientID
```

Comment:
Would it be better to always start file names with vendor and class so that they're nicely grouped in the ls output?

Reply:
True. I'll update it like that.